routing/bandwidth.go

```diff
@@ -85,22 +85,19 @@ func newBandwidthManager(graph Graph, sourceNode route.Vertex,
 // queried is one of our local channels, so any failure to retrieve the link
 // is interpreted as the link being offline.
 func (b *bandwidthManager) getBandwidth(cid lnwire.ShortChannelID,
-	amount lnwire.MilliSatoshi) lnwire.MilliSatoshi {
+	amount lnwire.MilliSatoshi) (lnwire.MilliSatoshi, error) {
 
 	link, err := b.getLink(cid)
 	if err != nil {
-		// If the link isn't online, then we'll report that it has
-		// zero bandwidth.
-		log.Warnf("ShortChannelID=%v: link not found: %v", cid, err)
-		return 0
+		return 0, err
 	}
 
 	// If the link is found within the switch, but it isn't yet eligible
 	// to forward any HTLCs, then we'll treat it as if it isn't online in
 	// the first place.
 	if !link.EligibleToForward() {
-		log.Warnf("ShortChannelID=%v: not eligible to forward", cid)
-		return 0
+		return 0, fmt.Errorf("failed to get bandwidth for channel %v: "+
```

Comment:
nice, I like the error approach better, but we don't need to add the channel id in the error messages, since those are already logged one layer up

Reply:
true updated